JumpRoyale/src/TimerOverlay.cs

```diff
@@ -0,0 +1,62 @@
+using System.Threading.Tasks;
+using Godot;
+
+public partial class TimerOverlay : VFlowContainer
+{
+    private const string SpriteName = "Sprite2D";
+    private const string TimerNodeName = SpriteName + "/Timer";
+
+    private static readonly int GameLength = GameOverlay.GameLength;
+
+    private int _timerSeconds = GameLength;
+
+    [Signal]
+    public delegate void TimerDoneEventHandler();
+
+    public void Init()
+    {
+        _ = StartTimer();
+    }
+
+    public async Task StartTimer()
+    {
+        await ToSignal(GetTree().CreateTimer(1.0f), SceneTreeTimer.SignalName.Timeout);
+
+        UpdateTimer();
+
+        if (_timerSeconds <= 0)
+        {
+            EmitSignal(SignalName.TimerDone);
+        }
+        else
+        {
+            _ = StartTimer();
+        }
```

Comment:
Usually, with if-else blocks (or nested `if`s) I think it's better to do an early return (where possible) instead of falling into else blocks to reduce nesting and to keep things at low-indent. With more conditions it becomes more readable and makes it more clear what each section does and early return just tells us that we don't have to pay attention to the rest of the code. This is just a nitpick and will make a bit more sense in the `UpdateTimer()`

```suggestion
        if (_timerSeconds <= 0)
        {
            EmitSignal(SignalName.TimerDone);
            return;
        }

        _ = StartTimer();
```